### ISSUE TITLE
Behavior of signed division operator on type Integer doesn't match documentation

### ISSUE THREAD

**Issue by brianhuffman** [GaloisInc/cryptol#661]
```
Cryptol> :? /$

    (/$) : {a} (Arith a) => a -> a -> a

Precedence 90, associates to the left.

2's complement signed division.  Division rounds toward 0.

Cryptol> (-5 : Integer) /$ 2
-3
Cryptol> (5 : Integer) /$ -2
-3
```
Contrary to the documentation, it appears that `/$` on `Integer` rounds down instead of toward 0.

On the other hand, for type `[n]`, it appears that `/$` does round toward 0:
```
Cryptol> (-5 : [8]) /$ 2 == -2
True
Cryptol> (5 : [8]) /$ -2 == -2
True
```
It seems that on type `Integer`, `/$` is just implemented as `/`, both in the evaluator and proof backends:
```
Cryptol> :check \(x : Integer) y -> y != 0 ==> x / y == x /$ y
Using random testing.
Passed 100 tests.
Cryptol> :prove \(x : Integer) y -> y != 0 ==> x / y == x /$ y
Q.E.D.
(Total Elapsed Time: 0.031s, using Z3)
```

**Comment by brianhuffman** [comment]
Operator `/$` probably shouldn't even be applicable to type `Integer` (see comments on #686). Restricting the type of `/$` to exclude `Integer` would make this problem go away.

**Comment by robdockins** [comment]
Fixed via #724. These operators not longer apply to `Integer`
